From 2bfce5937017bb8058d9093ac2249afc50b92813 Mon Sep 17 00:00:00 2001
From: Eldar Abusalimov <eldar.abusalimov@jetbrains.com>
Date: Tue, 24 Oct 2017 13:48:31 +0300
Subject: [PATCH 3/4] python: Make compatible with Python 3 w.r.t. integer
 division

# Conflicts:
#	examples/synthetic/gnu_libstdcpp.py
#	scripts/Python/modify-python-lldb.py
---
diff --git a/bindings/interface/SBDataExtensions.i b/bindings/interface/SBDataExtensions.i
--- a/bindings/interface/SBDataExtensions.i
+++ b/bindings/interface/SBDataExtensions.i
@@ -167,7 +167,7 @@ public:
                 else:
                     return my_data
             def __len__(self):
-                return int(self.sbdata.GetByteSize()/self.item_size)
+                return int(self.sbdata.GetByteSize()//self.item_size)
             def all(self):
                 return self[0:len(self)]

diff --git a/bindings/python/python.swig b/bindings/python/python.swig
--- a/bindings/python/python.swig
+++ b/bindings/python/python.swig
@@ -5,6 +5,9 @@
    functions for various scripting languages, to enable them to call the
    liblldb Script Bridge functions.
 */
+%pythonbegin %{
+from __future__ import division
+%}

 /* Define our module docstring. */
 %define DOCSTRING
diff --git a/examples/synthetic/libcxx.py b/examples/synthetic/libcxx.py
--- a/examples/synthetic/libcxx.py
+++ b/examples/synthetic/libcxx.py
@@ -101,7 +101,7 @@ class stdvector_SynthProvider:
             if (num_children % self.data_size) != 0:
                 return 0
             else:
-                num_children = num_children / self.data_size
+                num_children = num_children // self.data_size
             return num_children
         except:
             return 0

-- 
2.23.0
